### PR TITLE
code_80521D0.s (dungeon_message.c and dungeon_move.c)

### PR DESCRIPTION
**Pull request by DizzyEggg**
WIP

### CODE REVIEW

**Comment by DizzyEggg** [comment]
I decided to rename the file to `dungeon_message.c` as it's all dungeon string related functions.

I also named/renamed bunch of stuff, so any critique is welcome 😄

File: include/makuhita_dojo1.h [hunk 1]
```diff
@@ -28,15 +28,8 @@ typedef struct MakuhitaDojoWork1
...
-    /* 0x5C */ u8 *faceData;
-    u16 unk60;
-    u16 unk62;
-    u8 unk64;
-    u8 unk65;
-    u8 unk66;
-    u8 unk67;
-    OpenedFile** unk68;
+    /* 0x58 */ struct MonPortraitMsg monPortrait;
+    struct MonPortraitMsg* unk68;
```
Comment (SethBarberee) [inline include/makuhita_dojo1.h:32]:
```suggestion
    struct MonPortraitMsg* monPortraitPtr;
```

**Comment by DizzyEggg** [comment]
Ready for review.
Two new files created from code_80521D0.s. Lots of renaming, documentation, EWRAM/const variables moving and clean-up.

File: src/dungeon_move.c [hunk 2]
```diff
@@ -0,0 +1,3314 @@
...
+                }
+
+                if ((GetMoveTargetAndRangeForPokemon(attacker, move, FALSE) & 0xF) == 0) {
+                    SetShopkeeperAggression(attacker, currTarget);
+                }
+                targetInfo->unk158 = 1;
+
+                // They really must've had a soft spot for all these giant switches
+                switch (moveId) {
+                    case 9:
```
Comment (AnonymousRandomPerson) [inline src/dungeon_move.c:672]: Looks like these cases could all use move ID macros. Maybe making a quick replace script would be easier with the sheer number of cases.

File: include/dungeon_message.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    s16 unk6;
+    const u8 *str;
+};
+
+void sub_80521D0(void);
+void sub_8052210(bool8 a0);
+
+// These functions display the string if certain conditions are met. The conditions differ depending on the function. The string is then saved to the message log.
+void TryDisplayDungeonLoggableMessage(Entity *pokemon, const char *str);
+void TryDisplayDungeonLoggableMessage2(Entity *r0, const char *str);
```
Comment (SethBarberee) [inline include/dungeon_message.h:28]: Should probably keep the same standard of `r0` -> `pokemon` and `r1` -> `target` for these funcs like all the rest.

Comment (DizzyEggg) [reply include/dungeon_message.h:28]: Good idea

File: src/dungeon_move.c (hunk 2 above)
Comment (DizzyEggg) [reply src/dungeon_move.c]: Chat GPT wrote a cool script which did the job :D

**Comment by DizzyEggg** [comment]
Reviews applied

**Review by AnonymousRandomPerson — APPROVED** [review]

**Pull request merged**